src/msg/async/ProtocolV2.cc

```diff
@@ -796,7 +796,7 @@ CtPtr ProtocolV2::read(CONTINUATION_RXBPTR_TYPE<ProtocolV2> &next,
 }
 
 template <class F>
-CtPtr ProtocolV2::write(const std::string &desc,
+CtPtr ProtocolV2::write(const char *desc,
```

Comment:
```suggestion
CtPtr ProtocolV2::write(std::string_view desc,
```

better would be ^ and fix calls to `WRITE` to pass a `std::string_view` literal, e.g.:

```
return WRITE(reset, "session reset"sv, read_frame);
```

Good eye spotting this inefficiency.

Reply:
I don't agree that `std::string_view` should be used here. Sometimes, `std::string_view` is more efficient, but not in this case. Passing it consumes two registers instead of one, and the string length is only ever used when a connection-fatal I/O error occurs. The advantages of `std::string_view` will materialize extremely rarely, but you always carry its overhead around (twice).

Reply:
Also note that `std::string_view` literals have zero advantage over implicitly casting a C string literal to `std::string_view`. The compiler will do the implicit cast at compile time.

Reply:
> I don't agree that `std::string_view` should be used here. Sometimes, `std::string_view` is more efficient, but not in this case. Passing it consumes two registers instead of one, and the string length is only ever used when a connection-fatal I/O error occurs. The advantages of `std::string_view` will materialize extremely rarely, but you always carry its overhead around (twice).

Sorry, this code path is just not performance sensitive enough for me to entertain register spills as a reason to not use an applicable non-trivial type. We're not programming the kernel here.

In any case, passing the string_view **does** mean that the length is known wherever it is used (like debug prints). That's an attractive property and is a good reason for us to, by default, lean towards using `std::string_view`.

Reply:
> Also note that `std::string_view` literals have zero advantage over implicitly casting a C string literal to `std::string_view`. The compiler will do the implicit cast at compile time.
> 
> Proof: https://godbolt.org/z/55dseMfxe

That's good to know! Thanks for checking that.

Reply:
> Sorry, this code path is just not performance sensitive enough for me to entertain register spills as a reason to not use an applicable non-trivial type.

I changed the patch to use `std::string_view` instead.